downstairs/src/region.rs

```diff
@@ -1070,12 +1087,69 @@ impl Region {
         &self,
         flush_number: u64,
         gen_number: u64,
+        snapshot_details: &Option<SnapshotDetails>,
     ) -> Result<(), CrucibleError> {
+        // It should be ok to Flush a read-only region, but not take a snapshot.
```

Comment:
Hmm, thinking about it now, maybe should be illegal, or a panic.

If you have a volume with a read only parent, and a rw portion, and you take a snapshot, the snapshot should land on the rw portion only. Assuming a migration job is running, the snapshot will be inserted somewhere in the middle of the block copies. If the migration isn't done, anything booting off that snapshot needs the same read only parent.

Unfortunately I don't think we can get away with blocking.

Reply:
Hm.. yeah, this could get tricky.

Maybe we start by not allowing a snapshot of a volume with a RO parent until that volume has completed migration?